### DISCUSSION TITLE
TextBox on Android not working at all

### DISCUSSION THREAD

**Discussion by lukaszossowski** [AvaloniaUI/Avalonia#9423] [Q&A]
Hi,

I need to get text input working on mobile, specifically Android. I've used a textbox for this:

`				<TextBox MaxLines="1" TextAlignment="Center" Text={Binding Name} />`

and when I press the textbox it brings up the keyboard. However now there come many issues:

1. Name property has a default value ("Comment") that shows up in the textbox.  When I press any key on the keyboard it copies the default value and then adds the new sign leading to result like this after pressing "b" - "CommentCommentb".
2. When I use the backspace button, it deletes the Commentb part and then places cursor at the beginning of the textbox, but the initial "Comment" still remains. It is ignored until I place the cursor again at its end and then I can delete it.
3. Pressing space deletes existing text and then it enters a weird mode where any new key I press deletes the last one, so my textbox's text now only displays the last letter I pressed and nothing more. The mode works in a way that after pressing a letter I need to press space again to "accept" it and then it moves the cursor one space and now the second letter is displayed as the last key I pressed - until I accept it with spacebar and move on.
4. Pressing OK on the keyboard does nothing, there is no way to actually accept the text that is written.

Overall the experience is mind-boggingly broken and I have no idea what is happening. 

Am I doing something wrong or are those known issues? Has anyone had success with properly working textbox on mobile?

**Comment by Gillibald** [comment]
What version of Avalonia are you using?

**Reply by lukaszossowski** [reply]
11 preview 2
My phone is S20 Ultra

**Reply by Gillibald** [reply]
Please try preview4

**Reply by lukaszossowski** [reply]
It does work better on preview 4, thank you. Accepting input works ok. However when I use space it still jumps to weird places in the text and replicates behaviour from point 3 of my original question. Behaviour from point 1 also still persists.

**Reply by Gillibald** [reply]
Y I can repro the issue with typing space and caret positioning.

**Reply by lukaszossowski** [reply]
Should I create an issue about this?
